vectorsearch/resource_vector_search_endpoint.go

```diff
@@ -47,6 +48,13 @@ func ResourceVectorSearchEndpoint() common.Resource {
 			}
 			endpoint, err := wait.GetWithTimeout(d.Timeout(schema.TimeoutCreate))
 			if err != nil {
+				ctx2, cancel := context.WithDeadline(context.Background(), time.Now().Add(10*time.Second))
+				log.Printf("[ERROR] Error waiting for endpoint to be created: %s", err.Error())
+				err2 := w.VectorSearchEndpoints.DeleteEndpointByEndpointName(ctx2, req.Name)
```

Comment:
Is this behaviour consistent? i.e. deleting works even if creation fails? This sounds strange and probably should be fixed on backend. Maybe it is similar to cancelling while creation is in progress. I think we should confirm this with the API owning team.